File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"

uint16_t color_code(resistor_band_t color){
    return (uint16_t)color;
}

const resistor_band_t* colors(){
    static const resistor_band_t all_colors[] = {
        BLACK, BROWN, RED, ORANGE, YELLOW,
        GREEN, BLUE, VIOLET, GREY, WHITE
    };
    return all_colors;
}

File: solutions/python/resistor-color/1/resistor_color.py
"""
To match color and to return the index
"""
def color_code(color):
    """
    :param color:str -指定的颜色
    :return :int -对应的值
    """
    color_collection = colors()

    return color_collection.index(color)

def colors():
    return [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    ]
